signer/src/transaction_signer.rs

```diff
@@ -669,19 +615,22 @@ where
             // The compiler will complain about this, so we silence the
             // warning.
             #[allow(clippy::map_entry)]
-            WstsNetMessage::NonceRequest(_) => {
+            WstsNetMessage::NonceRequest(request) => {
                 if !chain_tip_report.sender_is_coordinator {
                     tracing::warn!("Got coordinator message from wrong signer");
                     return Ok(());
                 }
-                // TODO(296): Validate that message is the appropriate sighash
+
+                let db = self.context.get_storage();
+                Self::validate_bitcoin_sign_request(&db, &request.message).await?;
```

Comment:
Using the specific sighash is enough, since it is very unlikely that someone can use a particular sighash for a different transaction ID. That said, it might be useful to match on the transaction ID as well to ensure that the coordinator sends the actual transaction ID and not some random stuff. Right now it is used as an "identifier" for the stuff being signed but not necessarily the transaction ID.

Reply:
Got it, thanks! Since it’s not a security risk, the current approach seems fine as is. 👍